### PR TITLE
Bug 1832066 Initialize nimbus-cli tooling

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [Bug 1832066](https://bugzilla.mozilla.org/show_bug.cgi?id=1832066).

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1832066

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 1]
```diff
@@ -221,6 +222,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         // DO NOT MOVE ANYTHING ABOVE THIS getProfilerTime CALL.
         val startTimeProfiler = components.core.engine.profiler?.getProfilerTime()
 
+        // Setup nimbus-cli tooling. This is a NOOP when launching normally.
```
Comment (jonalmeida) [inline fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:225]: From the filed issue:

> ..the earliest time the app has access to an intent.

If this means an external intent, then maybe you want this in the `IntentReceivedActivity`? That is earliest time an intent is received.

File: focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt [hunk 2]
```diff
@@ -86,6 +87,7 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
         }
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        components.experiments.initializeTooling(applicationContext, intent)
```
Comment (jonalmeida) [inline focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:90]: Likewise for Focus too: IntentReceiverActivity might be what you want.

**Comment by jonalmeida** [comment]
The linked bug for the Focus change seems unrelated, I think? Is BZ-1832077, correct?

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt (hunk 1 above)
Comment (jhugman) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:225]: This is CLI sends a `android.intent.category.LAUNCHER` intent with arguments. This is to simulate as closely as possible launching the app by tapping the app icon from the homescreen/springboard/launcher.

Can I/should I send a `LAUNCHER` intent to `IntentReceiverActivity` instead? WDYT.

**Comment by jhugman** [comment]
> The linked bug for the Focus change seems unrelated, I think? Is [BZ-1832077](https://bugzilla.mozilla.org/show_bug.cgi?id=1832077), correct?

You're right! I had tried to remove it, but I'm not sure what I've done wrong to attract it in the first place, or to keep it coming back.

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt (hunk 1 above)
Comment (jonalmeida) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:225]: Gotcha, that sounds fine then!

**Review by jonalmeida — APPROVED** [review]

**Comment by jhugman** [comment]
@Mergifyio backport releases_v114

**Pull request merged**